Flair Indexing (#1)
0xflair/doc.zk-evm: docs/build-on-linea/tooling/data-indexers/flair.md

* ✅ **Custom processor scripts** with Javascript runtime (with **Typescript** support)
  * Make external API or Webhook calls to third-party or your backend.
  * Get current or historical USD value of any ERC20 token amount of any contract address on any chain.
  * Use any external NPM library.
* ✅ **Stream** any stored data to your destination database (Postgres, MognoDB, MySQL, Kafka, Elasticsearch, Timescale, etc).

## Getting Started

1️⃣ Clone the [starter boilerplate](https://github.com/flair-sdk/starter-boilerplate) template and follow the instructions:

```bash
git clone https://github.com/flair-sdk/starter-boilerplate.git
# ... follow instructions in README.md
```
::: info

Boilerplate instructions will create a **new cluster**, generate **an API Key**, and set up a manifest.yml to index your **first contract** with **sample custom processor** scripts.

Learn more about the [structure of manifest.yml](reference/manifest.yml.md).

:::

2️⃣ Configure linea RPC nodes

Set a unique namespace, Linea chainId and RPC endpoint in your `config`. Remember that you can add up to 10 RPC endpoints for resiliency.

```yaml
{
  "cluster": "dev",

>   "namespace": "linea-indexing-dev",

Reviewer: ```suggestion
  "namespace": "my-awesome-linea-indexing-dev",
```

  "indexers": [
    {
      "chainId": 59140,
      "enabled": true,
      "ingestionFilterGroup": "default",
      "processingFilterGroup": "default",
      "sources": [
        # Highly-recommended to have at least 1 websocket endpoint
        "wss://linea-mainnet.infura.io/v3/xxxxxxxxxx",
        "https://linea-mainnet.infura.io/v3/xxxxxxxxxx",
        # You can put multiple endpoints for failover
        "https://rpc.linea.build"
      ]
    }
  ]
}
```

> 3️⃣ [Query](reference/database.md) your custom indexed data.

Reviewer: ```suggestion
3️⃣ [Query](https://docs.flair.build/reference/database) your custom indexed data.
```

> 3️⃣ [Query](reference/database.md) your custom indexed data.

Reviewer: ```suggestion
3️⃣ [Query](https://docs.flair.build/#getting-started) your custom indexed data.
```

> 4️⃣ Stream the data to your [own database](reference/database.md).

Reviewer: ```suggestion
4️⃣ Stream the data to your [own database](https://docs.flair.build/reference/database).
```
